### ISSUE TITLE
About BPR Loss

### ISSUE THREAD

**Issue by whonor** [taoqi98/CAUM#3]
Hi there, 
In your paper, I find the loss is BPR loss, but the categorical cross-entropy loss is used in your code.
I think the two loss functions are different.
Why?

**Comment by taoqi98** [comment]
When the number (N) of negative samples is set to 1, the cross-entropy loss is equal to the BRP loss, because:
L = exp(s_p)/{ exp(s_p) + \sum_{k=1}^N exp(s^i_n) } = exp(s_p)/{ exp(s_p) + exp(s_n) } |[N=1] = 1/{1+exp(s_n-s_p)} = sigmoid(s_p-s_n), where s^i_n denotes the score of the i-th negative sample and s_p denotes the score of the positive sample.
In my released codes, I set N to 4 for better performance.
